recipes-browser/wpewebkit/wpewebkit_2.38.0.bb

```diff
@@ -32,3 +32,6 @@ PACKAGECONFIG[introspection] = "-DENABLE_INTROSPECTION=ON,-DENABLE_INTROSPECTION
 
 # Layer-Based SVG Engine
 PACKAGECONFIG[lbse] = "-DENABLE_LAYER_BASED_SVG_ENGINE=ON,-DENABLE_LAYER_BASED_SVG_ENGINE=OFF, "
+
+# unifdef-native: Needed since >2.38.
+DEPENDS:append:class-devupstream = " unifdef-native"
```

Comment:
@guijemont  can you verify if that works for you?

Reply:
It does not work. I guess it could work if I had a way to "enable" devupstream in local.conf? Without any specific modification to local.conf, here is the DEPENDS section of the output of `bitbake -e wpewebkit`:

```
# $DEPENDS [12 operations]
#   set /home/guijemont/dev/yocto-rpi-wpe/poky/meta/conf/bitbake.conf:305
#     ""
#   set /home/guijemont/dev/yocto-rpi-wpe/poky/meta/conf/documentation.conf:136
#     [doc] "Lists a recipe's build-time dependencies (i.e. other recipe files)."
#   :prepend /home/guijemont/dev/yocto-rpi-wpe/poky/meta/classes/base.bbclass:73
#     "${BASEDEPENDS} "
#   set /home/guijemont/dev/yocto-rpi-wpe/meta-webkit/recipes-browser/wpewebkit/wpewebkit.inc:14
#     "     ${@bb.utils.contains_any('LAYERSERIES_CORENAMES', 'dunfell gatesgarth hardknott honister', 'libsoup-2.4', 'libsoup', d)}     bison-native gperf-native harfbuzz-native libxml2-native ccache-native ninja-native ruby-native cairo 
    fontconfig freetype glib-2.0 harfbuzz icu jpeg pcre sqlite3 zlib libpng     libwebp libxml2 libxslt virtual/egl virtual/libgles2 libepoxy libgcrypt "
#   :prepend /home/guijemont/dev/yocto-rpi-wpe/poky/meta/classes/cmake.bbclass:4
#     "cmake-native "
#   :prepend /home/guijemont/dev/yocto-rpi-wpe/poky/meta/classes/pkgconfig.bbclass:1
#     "pkgconfig-native "
#   append /home/guijemont/dev/yocto-rpi-wpe/poky/meta/classes/perlnative.bbclass:2
#     "perl-native"
#   :append /home/guijemont/dev/yocto-rpi-wpe/poky/meta/classes/python3native.bbclass:5
#     " python3-native "
#   append /home/guijemont/dev/yocto-rpi-wpe/meta-webkit/recipes-browser/wpewebkit/wpewebkit_2.38.0.bb:13
#     " libwpe"
#   :append[class-devupstream] /home/guijemont/dev/yocto-rpi-wpe/meta-webkit/recipes-browser/wpewebkit/wpewebkit_2.38.0.bb:37
#     " unifdef-native"
#   append base.bbclass:489 [appendVar]
#     " gstreamer1.0-plugins-base gstreamer1.0 gstreamer1.0-plugins-good openjpeg gstreamer1.0 gstreamer1.0-plugins-base gstreamer1.0 gstreamer1.0-plugins-base gstreamer1.0-plugins-good libgcrypt libtasn1 woff2"
#   append cmake.bbclass:19 [__anon_24__home_guijemont_dev_yocto_rpi_wpe_poky_meta_classes_cmake_bbclass]
#     " ninja-native"
# pre-expansion value:
#   "pkgconfig-native cmake-native ${BASEDEPENDS}      ${@bb.utils.contains_any('LAYERSERIES_CORENAMES', 'dunfell gatesgarth hardknott honister', 'libsoup-2.4', 'libsoup', d)}     bison-native gperf-native harfbuzz-native libxml2-native ccache-native ninja-native ruby-native cairo     fontconfig freetype glib-2.0 harfbuzz icu jpeg pcre sqlite3 zlib libpng     libwebp libxml2 libxslt virtual/egl virtual/libgles2 libepoxy libgcrypt  perl-native  libwpe python3-native  gstreamer1.0-plugins-base gstreamer1.0 gstreamer1.0-plugins-good openjpeg gstreamer1.0 gstreamer1.0-plugins-base gstreamer1.0 gstreamer1.0-plugins-base gstreamer1.0-plugins-good libgcrypt libtasn1 woff2 ninja-native"
DEPENDS="pkgconfig-native cmake-native virtual/arm-poky-linux-gnueabi-gcc virtual/arm-poky-linux-gnueabi-compilerlibs virtual/libc      libsoup     bison-native gperf-native harfbuzz-native libxml2-native ccache-native ninja-native ruby-native cairo     fontconfig freetype glib-2.0 harfbuzz icu jpeg pcre sqlite3 zlib libpng     libwebp libxml2 libxslt virtual/egl virtual/libgles2 libepoxy libgcrypt  perl-native  libwpe python3-native  gstreamer1.0-plugins-base gstreamer1.0 gstreamer1.0-plugins-good openjpeg gstreamer1.0 gstreamer1.0-plugins-base gstreamer1.0 gstreamer1.0-plugins-base gstreamer1.0-plugins-good libgcrypt libtasn1 woff2 ninja-native"
```

Note that we have the `:append[class-devupstream]` step, but `unifdef-native` does not end up in `DEPENDS`.

Probably worth noting that in the output I also have `CLASSOVERRIDE="class-target"` (i.e. no `class-devupstream` here) and also we have `OVERRIDES="linux-gnueabi:arm:pn-wpewebkit:rpi:armv7ve:raspberrypi3:poky:class-target:libc-glibc:forcevariable"` (no `class-devupstream` there either).

It does seem to work if I add the following in my local.conf though:
```
CLASSOVERRIDE:append:pn-wpewebkit = ":class-devupstream"
```

Then in the output of `bitbake -e wpewebkit` I get `class-devupstream` showing in `OVERRIDES` and `CLASSOVERRIDE` and `unifdef-native` is in `DEPENDS`.

Reply:
To modify the `local.conf` by adding `include conf/include/webkit-bleeding.inc` is expected.  I wonder why you don't want to modify the local.conf when indeed is the way to say to the build env that you are going to build devupotream versions not the stable release.

Reply:
> To modify the `local.conf` by adding `include conf/include/webkit-bleeding.inc` is expected. I wonder why you don't want to modify the local.conf when indeed is the way to say to the build env that you are going to build devupotream versions not the stable release.

Sorry. I am fine to modify my local.conf. I did not know that I should include webkit-bleeding.inc from there, I will try that next. I suggest we should add a note about the recommended way of working with upstream in [README.md](https://github.com/Igalia/meta-webkit/blob/main/README.md) (in another patch).

My issue was with adding  the line I quoted in my local.conf, which felt a little hackish (though that could be OK if documented as the recommended way of doing things).

Reply:
Looks good an reasonable.

Reply:
Issue created for this purpose: https://github.com/Igalia/meta-webkit/issues/426

Reply:
BTW, I think that we would need to add this to [webkit-bleeding.inc](https://github.com/Igalia/meta-webkit/blob/main/conf/include/webkit-bleeding.inc):
```
CLASSOVERRIDE:append:pn-wpewebkit = ":class-devupstream"
```
And probably something similar for cog and maybe other projects, so that the devupstream stuff is selected when including that file. I don't think it would be included as the file is right now.

An alternative could be to add it in [devupstream.inc](https://github.com/Igalia/meta-webkit/blob/main/conf/include/devupstream.inc), which would make more sense given the name of the file, but that would change from its current meaning (currently, it is included in the release recipes such as [wpewebkit_2.38.2.bb](https://github.com/Igalia/meta-webkit/blob/main/recipes-browser/wpewebkit/wpewebkit_2.38.2.bb), so adding the line to devupstream.inc means always enabling the devupstream stuff, which I think is confusing).

Reply:
That is done by `poky/meta/classes-recipe/devupstream.bbclass:    d.appendVar("CLASSOVERRIDE", ":class-devupstream")` thought  `require conf/include/devupstream.inc` added in the recipes what extends the recipes `BBCLASSEXTEND += "devupstream:target`